lib/ffmpeg/qsv/encoder.py

```diff
@@ -57,13 +60,165 @@ def inner(quality):
+        _encparams = f"{_encparams}:MaxKbps={_maxrate}"
+    if self.refs != None and len(self.refs.strip())>0:
+        _refs=re.findall("\d+", self.refs)[0]
+        _encparams = f"{_encparams}:NumRefFrame={_refs}"
+    if self.ladepth != None and len(self.ladepth.strip())>0:
+        _ladepth=re.findall("\d+", self.ladepth)[0]
+        _encparams = f"{_encparams}:LookAheadDepth={ladepth}"
+    if self.intref != None and len(self.intref.strip())>0:
+        # for exampe: self.intref="  -int_ref_type 1 -int_ref_cycle_size 4 -int_ref_cycle_dist 20"
+        # to get intref_type
+        r,_,_intref_type=self.intref.partition('-int_ref_type')
+        _intref_type,_,r=_intref_type.partition('-int_ref_cycle_size')
+        # to get intref_size
+        _intref_type=_intref_type.strip()
+        r,_,_intref_size=self.intref.partition('-int_ref_cycle_size')
+        _intref_size,_,r=_intref_size.partition('-int_ref_cycle_dist')
+        _intref_size=_intref_size.strip()
+        # to get intref_dist
+        r,_,_intref_refdist=self.intref.partition('-int_ref_cycle_dist')
+        _intref_refdist=_intref_refdist.strip()
+        _encparams = f"{_encparams}:mfxExtCodingOption2.IntRefType={_intref_type}:mfxExtCodingOption2.IntRefCycleSize={_intref_size}:mfxExtCodingOption3.IntRefCycleDist={_intref_refdist}"
+    if self.extbrc != None and len(self.extbrc.strip())>0:
+        _extbrc=re.findall("\d+", self.extbrc)[0]
+        _encparams = f"{_encparams}:ExtBRC={_extbrc}"
+    if self.ldb != None and len(self.ldb.strip())>0:
+        _ldb=re.findall("\d+", self.ldb)[0]
+        _encparams = f"{_encparams}:mfxExtCodingOption3.LowDelayBRC={_ldb}"
+    if self.tilecols != None and len(self.tilecols.strip())>0:
+        _tilecols=re.findall("\d+", self.tilecols)[0]
+        _encparams = f"{_encparams}:mfxExtAV1TileParam.NumTileColumns={_tilecols}"
+    if self.tilerows != None and len(self.tilerows.strip())>0:
+        _tilerows=re.findall("\d+", self.tilerows)[0]
+        _encparams = f"{_encparams}:mfxExtAV1TileParam.NumTileRows={_tilerows}"
+    if self.strict != None and len(self.strict.strip())>0:
+        _strict=re.findall("\d+", self.strict)[0]
+        _encparams = f"{_encparams}:GopOptFlag={_strict}"
+    if self.pict != None and len(self.pict.strip())>0:
+        _pict=re.findall("\d+", self.pict)[0]
+        _encparams = f"{_encparams}:mfxExtCodingOption.PicTimingSEI={_pict}"
+    if self.quality != None and len(self.quality.strip())>0:
+        _quality=re.findall("\d+", self.quality)[0]
+        _encparams = f"{_encparams}:TargetUsage={_quality}"
+    if self.lowpower != None and len(self.lowpower.strip())>0:
+        _lowpower=re.findall("\d+", self.lowpower)[0]
+        _encparams = f"{_encparams}:LowPower={_lowpower}"
+    #{self.forced_idr} ffmpeg
+    #{self.pict}{self.rqp}
+
+    if _encparams != None and len(_encparams.strip()) > 1:
+        if ':' == _encparams[0]:
+            _encparams = _encparams[1:]
```

Comment:
`_encparams` will never be `None`, so this could be `_encparams = _encparams.lstrip(':')`.

That said, how about make _encparams a list instead... append each option as needed, then just do `return ':'.join(_encparams)`

Reply:
sure, thanks 

> `_encparams` will never be `None`, so this could be `_encparams = _encparams.lstrip(':')`.
> 
> That said, how about make _encparams a list instead... append each option as needed, then just do `return ':'.join(_encparams)`

Sure, thanks suggestion